go-controller/pkg/testing/libovsdb/libovsdb.go

```diff
@@ -139,7 +140,7 @@ func newOVSDBTestHarness(serverData []TestData, newServer serverBuilderFn, newCl
 
 func newNBClient(cfg config.OvnAuthConfig, cleanup *Cleanup) (libovsdbclient.Client, error) {
 	stopChan := make(chan struct{})
-	libovsdbOvnNBClient, err := libovsdb.NewNBClientWithConfig(cfg, stopChan)
+	libovsdbOvnNBClient, err := libovsdb.NewNBClientWithConfig(cfg, prometheus.NewRegistry(), stopChan)
```

Comment:
I am wondering here, don't we need a stub for testing?

Reply:
I am not aware of any stub.
I didn't see the point of stubbing this out. Creating a new registry is cheap.
I can create one if you want me to.

Reply:
> I am not aware of any stub. I didn't see the point of stubbing this out. Creating a new registry is cheap. I can create one if you want me to.

Nah, I was just wondering what happens in the tests if there is no prometheus

Reply:
A little extra cpu / ram but no effect if theres no prometheus endpoint. It registers the metrics with the prometheus registry, and allows the metrics to be updated, but they are not served because theres no server serving the registry.